### DISCUSSION TITLE
TO Modify static final fields

### DISCUSSION THREAD

**Discussion by PlainsVillager** [FabricMC/fabric-api#2850] [Mod Dev Support] [answered]
i want to edit the `public static final` field in Items.java with Mixin. But i dont know how to do.
here are the codes i want to replace in Items.java
`public static final Item SNOWBALL = new Item(new Item.Settings());`

**Comment by KosmX** [comment, marked as answer]
To modify static, you can inject into a special function `clinit` *class-init*:  
```java
	@Mutable
	@Shadow @Final public static Item SNOWBALL;

	@Inject(method = "<clinit>", at = @At("TAIL"))
	private static void modifySnowball(CallbackInfo ci) {
		SNOWBALL = /* ... */
	}
```
But I don't think, changing a built-in item this way is a good idea... I would try redirecting the snowball item constructor call.

**Reply by PlainsVillager** [reply]
may this modify field?

**Reply by PlainsVillager** [reply]
So the principle is to inject a method to set the value of the field?

**Reply by PlainsVillager** [reply]
I should use *might* to ask

**Reply by PlainsVillager** [reply]
why it didnt work here

**Reply by KosmX** [reply]
why? - very meaningful comments :D

So, as I said before, modifying the final static this way in Items class works (the mixin works) but the game crashes due to an inconsistent state.

Instead use constructor redirecting to *modify* the item **before** it can be added to a registry:
```java
	@Redirect(method = "<clinit>", at = @At(value = "NEW", target = "net/minecraft/item/SnowballItem"))
	private static SnowballItem modifySnowball(Item.Settings settings) {
		return new SnowballItem(new Item.Settings().maxCount(32)); // modify stack size to 32
	}
```
  
Another solution is to modify the argument of `Items.register`

**Reply by PlainsVillager** [reply]
thanks it worked
